### PR TITLE
Php1branch > tl1revbranch

### PR DESCRIPTION
**Pull request by php-1**

### CODE REVIEW

File: portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php [hunk 1]
```diff
@@ -142,31 +142,25 @@
 
     if (! $error)
     { 
-      if (isset($_POST["code"]) && preg_match("/^[a-z0-9-_]+$/",$_POST['code']) && !is_numeric($_POST["code"]))
+
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php]: why the whole condition removed?
at least keep (isset($_POST["code"]))

File: portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php [hunk 2]
```diff
@@ -142,31 +142,25 @@
...
+            header("Location: ".$_SERVER["PHP_SELF"]);
+            exit;
         }
         else
         {
             $error++;
-            $langs->load("errors");
-            $mesg=$langs->trans("ErrorFieldCanNotContainSpecialCharacters",$langs->transnoentities("code"));
-            setEventMessages($mesg, null, 'errors');
+            setEventMessages($langs->trans('Updated User relative type'), null, 'errors');
```
Comment (hythamsoliman81) [inline portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php]: why using spacing at "Updated User relative type"?

File: portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php (hunk 1 above)
Comment (php-1) [reply portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php]: ==> this condition check on code that's enough
if (empty($codetype))
{
$error++;
$langs->load("errors");
$mesg[]=$langs->trans("ErrorFieldRequired",$langs->transnoentitiesnoconv("code type user relative "));
$action = 'edit';
}

File: portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php (hunk 2 above)
Comment (php-1) [reply portfolio/ifintech/v1/ierp/htdocs/custom/ihcm/admin/typeuserrelative.php]: this file not completed the translate in folder lang

**Pull request merged**